tests/tools/test_flake8_plugin.py

```diff
@@ -232,3 +232,64 @@ def test_S013():
 """
     expected = ["t.py:1:0: S013 Use `django.contrib.postgres.fields.array.ArrayField` instead"]
     assert _run(src) == expected
+
+
+def test_S014_catches_global_assignment():
+    bad = """\
+from concurrent.futures import ThreadPoolExecutor
+
+# 1 worker each for spans, errors, performance issues
+_query_thread_pool = ThreadPoolExecutor(max_workers=3)
+"""
+    expected = ["t.py:4:21: S014 All `ThreadPoolExecutor` must have a `thread_name_prefix`."]
+    assert _run(bad) == expected
+
+    good = """\
+from concurrent.futures import ThreadPoolExecutor
+
+# 1 worker each for spans, errors, performance issues
+_query_thread_pool = ThreadPoolExecutor(max_workers=3, thread_name_prefix='xyz')
+"""
+    assert _run(good) == []
+
+
+def test_S014_catches_module_import():
+    bad = """\
+import concurrent.futures
+
+# 1 worker each for spans, errors, performance issues
+_query_thread_pool = concurrent.futures.ThreadPoolExecutor(max_workers=3)
+"""
+    expected = ["t.py:4:21: S014 All `ThreadPoolExecutor` must have a `thread_name_prefix`."]
+    assert _run(bad) == expected
+
+    good = """\
+import concurrent.futures
+
+# 1 worker each for spans, errors, performance issues
+_query_thread_pool = concurrent.futures.ThreadPoolExecutor(thread_name_prefix='abc', max_workers=3)
+"""
+    assert _run(good) == []
+
+
+def test_S014_immediate_with_is_okay():
+    okay = """\
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+class MyClass:
+    def my_method(self):
+        # Use a threadpool to send requests concurrently
+        with ThreadPoolExecutor(max_workers=worker_threads) as threadpool:
+            ...
+"""
+    assert _run(okay) == []
+
+
+def test_S014_checks_body_of_with():
+    notsure = """\
```

Comment:
?

Reply:
I wasn't sure, at writing, then forgot...
Actually, I'm still not sure, a bit. Your earlier review indicated this test should fail, but it's apparently passing.
Renamed to "okay".